src/watertap_contrib/seto/unit_models/surrogate/lt_med_surrogate.py

```diff
@@ -356,354 +374,396 @@ def temp_last_effect(b):
                     + temp_steam**3
-                    * b.recovery_vol_phase[0, "Liq"]
-                    * specific_area_coeffs[b.number_effects.value][63]
+                    * self.recovery_vol_phase[0, "Liq"]
+                    * specific_area_coeffs[num_effect][63]
                     + temp_steam**3
-                    * b.temp_last_effect
-                    * specific_area_coeffs[b.number_effects.value][64]
-                    + 1 * specific_area_coeffs[b.number_effects.value][65]
-                    + temp_steam**4 * specific_area_coeffs[b.number_effects.value][66]
-                    + b.temp_last_effect**4
-                    * specific_area_coeffs[b.number_effects.value][67]
-                    + b.recovery_vol_phase[0, "Liq"] ** 4
-                    * specific_area_coeffs[b.number_effects.value][68]
-                    + feed_conc_ppm**4
-                    * specific_area_coeffs[b.number_effects.value][69]
+                    * self.temp_last_effect
+                    * specific_area_coeffs[num_effect][64]
+                    + 1 * specific_area_coeffs[num_effect][65]
+                    + temp_steam**4 * specific_area_coeffs[num_effect][66]
+                    + self.temp_last_effect**4 * specific_area_coeffs[num_effect][67]
+                    + self.recovery_vol_phase[0, "Liq"] ** 4
+                    * specific_area_coeffs[num_effect][68]
+                    + feed_conc_ppm**4 * specific_area_coeffs[num_effect][69]
+                )
+
+        # Surrogate equations were built for 3,6,9,12,14 effects
+        # For intermediate number of effects (4,5,7,8,10,11,13), linear interpolation is adopted
+        @self.Constraint(doc="Gain output ratio surrogate equation")
+        def eq_gain_output_ratio(b):
+            if b.config.number_effects in [3, 6, 9, 12, 14]:
+                return b.gain_output_ratio == _get_gain_output_ratio(
+                    b.config.number_effects
+                )
+            elif b.config.number_effects in [4, 5, 7, 8, 10, 11, 13]:
+                # find out the closest numbers of effects that have a surrogate equation
+                interp_effects, i = [3, 6, 9, 12, 14], 1
+                while interp_effects[i] < b.config.number_effects:
+                    i += 1
+                interp_point1, interp_point2 = interp_effects[i - 1], interp_effects[i]
+
+                # implement linear interpolation using 2 points
+                return b.gain_output_ratio == _get_gain_output_ratio(interp_point1) * (
+                    interp_point2 - b.config.number_effects
+                ) / (interp_point2 - interp_point1) + _get_gain_output_ratio(
+                    interp_point2
+                ) * (
+                    b.config.number_effects - interp_point1
+                ) / (
+                    interp_point2 - interp_point1
+                )
+            else:
+                raise Exception(
```

Comment:
Should raise ``ConfigurationError`` and be checked at the beginning of ``build()`` I'd say

Reply:
Since you already of the domain set for number_effects, you should add a test to check what happens (whether you get exception as expected and what the error message is). You could then modify as needed to raise a relevant exception

Reply:
I added a test for the `ValueError` that is raised by IDAES `ConfigBlock` as Adam pointed out that it should be handled by IDAES already, but I also raised a `ConfigurationError` in the `build()` function as well (it shouldn't be necessary in the current setup as it won't be triggered).